apps/chat/channels.py

```diff
@@ -35,6 +37,21 @@
 You only support {supported_types} messages types. Respond only with the message for the user
 """
 
+# The regex from https://stackoverflow.com/a/6041965 is used, but tweaked to remove capturing groups
+URL_REGEX = r"(?:http|ftp|https):\/\/(?:[\w_-]+(?:(?:\.[\w_-]+)+))(?:[\w.,@?^=%&:\/~+#-]*[\w@?^=%&\/~+#-])"
+
+
+def strip_urls_and_emojis(text: str) -> tuple[str, list[str]]:
+    """Strips any URLs in `text` and appends them to the end of the text. Emoji's are filtered out"""
+    text = emoji.replace_emoji(text, replace="")
+
+    url_pattern = re.compile(URL_REGEX)
+    urls = url_pattern.findall(text)
+    for url in urls:
+        text = text.replace(url, "")
+
+    return text, urls
```

Comment:
you should deduplicate the urls: `list(set(urls))`

Reply:
I orignally did that, but thought "what if the bot references a URL now, then another two and then this one again"? Keeping the referenced URLs in order might help users a bit to know which URL the bot is referring to, but maybe we don't want to give users this potential cognative load where they have to keep track of where a url was referenced, then cross check that with the given list. Doesn't quite make sense. Back to deduplication!

Reply:
https://github.com/dimagi/open-chat-studio/pull/563/commits/9a13579bad6754d2269c6311be6fa3a182fc46f4

Reply:
I see your point. Either way it's probably unlikely that there will be duplicates.